src/services/redeem_and_unstake/ProveCoGatewayService.ts

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable import/no-unresolved */
+
+import assert from 'assert';
+import BigNumber from 'bignumber.js';
+import Web3 from 'web3';
+
+import { interacts } from '@openst/mosaic-contracts';
+import { ProofGenerator } from '@openst/mosaic-proof';
+
+import { ORIGIN_GAS_PRICE} from '../../Constants';
+import Logger from '../../Logger';
+import AuxiliaryChain from '../../models/AuxiliaryChain';
+import Observer from '../../observer/Observer';
+import GatewayRepository from '../../repositories/GatewayRepository';
+import {MessageDirection, MessageRepository} from '../../repositories/MessageRepository';
+import Utils from '../../Utils';
+import {EIP20Gateway} from "@openst/mosaic-contracts/dist/interacts/EIP20Gateway";
+
+export default class ProveCoGatewayService extends Observer<AuxiliaryChain> {
+  private gatewayRepository: GatewayRepository;
+
+  private messageRepository: MessageRepository;
+
+  private originWeb3: Web3;
+
+  private auxiliaryWeb3: Web3;
+
+  private originWorkerAddress: string;
+
+  private coGatewayAddress: string;
+
+  private auxiliaryChainId: number;
+
+  /**
+   *  Constructor
+   *
+   * @param gatewayRepository Instance of auxiliary chain repository.
+   * @param messageRepository Instance of message repository.
+   * @param originWeb3 Origin Web3 instance.
+   * @param auxiliaryWeb3 Auxiliary Web3 instance.
+   * @param originWorkerAddress auxiliary worker address, this should be
+   *                               unlocked and added in web3 wallet.
+   * @param gatewayAddress Address of gateway contract on aux chain.
```

Comment:
coGatewayAddress instead of gatewayAddresss

Reply:
Done!